lib/views/home.dart

```diff
@@ -209,6 +209,12 @@ class _MainScreenState extends State<MainScreen> with TickerProviderStateMixin {
                                             future: databaseFunctions
                                                 .fetchUserBeacons(),
                                             builder: (context, snapshot) {
+                                              if (snapshot.connectionState ==
+                                                  ConnectionState.waiting)
+                                                return Center(
+                                                  child:
+                                                      CircularProgressIndicator(),
+                                                );
```

Comment:
If connectionState.done is false then Card placeholder (with shimmer effect) is already there, which looks better and it would be better to add same while fetching nearbyBeacons instead of CircularProgressIndicator()

Reply:
oh yes, I will make changes and raise a PR again